src/extensions/sb-ncbr/partial-charges/property.ts

```diff
@@ -0,0 +1,193 @@
+
+    if (!typeIds || !atomIds || !charges) return atomIdToCharge;
+
+    for (let i = 0; i < rowCount; ++i) {
+        const typeId = typeIds[i];
+        const atomId = atomIds[i];
+        const charge = charges[i];
+        if (!atomIdToCharge.has(typeId)) atomIdToCharge.set(typeId, new Map());
+        atomIdToCharge.get(typeId)?.set(atomId, charge);
+    }
+
+    return atomIdToCharge;
+}
+
+function getTypeIdToResidueIdToCharge(model: Model, typeIdToAtomIdToCharge: ChargesData['typeIdToAtomIdToCharge']) {
+    const { offsets } = model.atomicHierarchy.residueAtomSegments;
+    const residueToCharge: ChargesData['typeIdToResidueToCharge'] = new Map();
+
+    typeIdToAtomIdToCharge.forEach((atomIdToCharge, typeId: number) => {
+        for (let residueId = 1; residueId < offsets.length; ++residueId) {
+            let charge = 0;
+            for (let atomId = offsets[residueId - 1] + 1; atomId <= offsets[residueId]; ++atomId) {
+                charge += atomIdToCharge?.get(atomId) || 0;
+            }
+            for (let atomId = offsets[residueId - 1] + 1; atomId <= offsets[residueId]; ++atomId) {
+                if (!residueToCharge.has(typeId)) residueToCharge.set(typeId, new Map());
+                residueToCharge.get(typeId)?.set(atomId, Number(charge.toFixed(4)));
+            }
+        }
+    });
+
+    return residueToCharge;
+}
+
+function getMaxAbsoluteCharges(
+    typeIdToCharge: ChargesData['typeIdToAtomIdToCharge']
+): ChargesData['maxAbsoluteAtomCharges'];
+function getMaxAbsoluteCharges(
+    typeIdToCharge: ChargesData['typeIdToResidueToCharge']
+): ChargesData['maxAbsoluteResidueCharges'] {
+    const maxAbsoluteCharges: Map<number, number> = new Map();
+
+    typeIdToCharge.forEach((idToCharge, typeId) => {
+        const charges = Array.from(idToCharge.values());
+        const min = Math.min(...charges);
+        const max = Math.max(...charges);
+        const bound = Math.max(Math.abs(min), max);
+        maxAbsoluteCharges.set(typeId, bound);
+    });
+
+    return maxAbsoluteCharges;
+}
+
+function getMaxAbsoluteAtomChargeAll(maxAbsoluteAtomCharges: ChargesData['maxAbsoluteAtomCharges']): number {
+    let maxAbsoluteCharge = 0;
+
+    maxAbsoluteAtomCharges.forEach((_, typeId) => {
+        const maxCharge = maxAbsoluteAtomCharges.get(typeId) || 0;
+        if (maxCharge > maxAbsoluteCharge) maxAbsoluteCharge = maxCharge;
+    });
+
+    return maxAbsoluteCharge;
+}
+
+function hasPartialChargesCategories(model: Model): boolean {
+    if (!MmcifFormat.is(model.sourceData)) return false;
+    const names = model.sourceData.data.frame.categoryNames;
+    return (
+        names.includes('atom_site') &&
+        names.includes('partial_atomic_charges') &&
+        names.includes('partial_atomic_charges_meta')
+    );
+}
+
+export function isApplicable(model?: Model): boolean {
+    return !!model && model.sourceData.kind === 'mmCIF' && hasPartialChargesCategories(model);
```

Comment:
The `model.sourceData.kind === 'mmCIF'` is already part of the `hasPartialChargesCategories`.